src/promptflow/promptflow/_sdk/_submitter/utils.py

```diff
@@ -156,19 +156,23 @@ def variant_overwrite_context(
     drop_node_variants: bool = False,
 ):
     """Override variant and connections in the flow."""
-    flow_dag_path, flow_dag = load_flow_dag(flow_path)
-    flow_dir_path = flow_dag_path.parent
-    if _get_additional_includes(flow_dag_path):
-        # Merge the flow folder and additional includes to temp folder.
-        with _merge_local_code_and_additional_includes(code_path=flow_path) as temp_dir:
-            # always overwrite variant since we need to overwrite default variant if not specified.
-            overwrite_variant(flow_dag, tuning_node, variant, drop_node_variants=drop_node_variants)
-            overwrite_connections(flow_dag, connections, working_dir=flow_dir_path)
-            overwrite_flow(flow_dag, overrides)
+    flow_dag = flow._data
+    flow_dir_path = Path(flow.code)
+    if flow.additional_includes:
+        # Merge the flow folder and additional includes to temp folder for both eager flow & dag flow.
+        with _merge_local_code_and_additional_includes(code_path=flow_dir_path) as temp_dir:
+            if not isinstance(flow, EagerFlow):
```

Comment:
this judgement looks a little wired. I mean, maybe it would be more clear if we divide additional includes handling logic and variant override logic?

Reply:
we can refactor this in further PR